frida_tools/apk.py

```diff
@@ -0,0 +1,313 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+
+
+from io import BufferedReader
+import struct
+from enum import IntEnum
+import os
+from zipfile import ZipFile
+
+
+class ChunkType(IntEnum):
+    STRING_POOL = 0x001
+    XML = 0x003
+    START_ELEMENT = 0x102
+
+
+class ResourceType(IntEnum):
+    BOOL = 0x12
+
+
+class StringType(IntEnum):
+    UTF8 = 1 << 8
+
+
+class BadHeader(Exception):
+    pass
+
+
+class ChunkHeader:
+    FORMAT = "<HHI"
+
+    def __init__(self, stream: BufferedReader, consume_data=True):
+        self.stream = stream
+        data = self.stream.peek(struct.calcsize(self.FORMAT))
+        (self.type, self.header_size, self.size) = struct.unpack_from(self.FORMAT, data)
+        if consume_data:
+            self.chunk_data = self.stream.read(self.size)
+        else:
+            self.chunk_data = self.stream.read(struct.calcsize(self.FORMAT))
+
+
+class StartElement:
+    FORMAT = "<HHIIIIIIHHHH"
+    ATTRIBUTE_FORMAT = "<IIiHBBi"
+
+    def __init__(self, header: ChunkHeader) -> None:
+        self.header = header
+        self.stream = self.header.stream
+        self.header_size = struct.calcsize(self.FORMAT)
+
+        data = struct.unpack_from(self.FORMAT, self.header.chunk_data)
+        if data[0] != ChunkType.START_ELEMENT:
+            raise BadHeader()
+
+        self.name = data[6]
+        self.attribute_count = data[8]
+
+        attributes_data = self.header.chunk_data[self.header_size:]
+        if len(attributes_data[-20:]) == 20:
+            previous_attribute = struct.unpack(
+                self.ATTRIBUTE_FORMAT, attributes_data[-20:])
+            self.namespace = previous_attribute[0]
+        else:
+            # There are no other attributes in the application tag
+            self.namespace = -1
+
+    def insert_debuggable(self, name: int) -> None:
+        # TODO: Instead of using the previous attribute to determine the probable
+        # namespace for the debuggable tag we could scan the strings section
+        # for the AndroidManifest schema tag
+        if self.namespace == -1:
+            raise BadHeader()
+
+        chunk_data = bytearray(self.header.chunk_data)
+
+        resource_size = 8
+        resource_type = ResourceType.BOOL
+        resource_data = -1  # True
```

Comment:
What does this `# True` comment try to explain? (Or is it an artifact left from an earlier iteration?)